source/StartConditionsPanel.cpp

```diff
@@ -0,0 +1,292 @@
+	
+	
+	const Font &font = FontSet::Get(14);
+	
+	Information info;
+	
+	// String that will be shown in the description panel
+	string descriptionText; 
+	
+	if(hasChosenStart) 
+	{
+		info.SetCondition("chosen start");
+		if(chosenStart.GetSprite())
+			info.SetSprite("start sprite", chosenStart.GetSprite());
+		info.SetString("name", chosenStart.GetName());
+		info.SetString("description", chosenStart.GetDescription());
+		info.SetString("planet", chosenStart.GetPlanet()->Name());
+		info.SetString("system", chosenStart.GetSystem()->Name());
+		info.SetString("date", chosenStart.GetDate().ToString());
+		info.SetString("credits", Format::Credits(chosenStart.GetAccounts().Credits()));
+
+		descriptionText = chosenStart.GetDescription();
+	}
+	else if(!GameData::Start().size())
+	{
+		descriptionText = "No start scenarios were defined!\n\nMake sure that you installed Endless Sky and all of your plugins properly";
+	}
+	
+	GameData::Background().Draw(Point(), Point());
+	GameData::Interfaces().Get("menu background")->Draw(info, this);
+	GameData::Interfaces().Get("start conditions menu")->Draw(info, this);
+	GameData::Interfaces().Get("menu start info")->Draw(info, this);
+	
+	
+	
+	// TODO: Prevent text from overflowing	
+	descriptionWrappedText.Draw(
+		Point(descriptionBox.Left(), descriptionBox.Top() + descriptionScroll),
+		bright
+	);
+	
+	Point point(entryListBox.Left(), entryListBox.Top() - listScroll);
+	
+	for(const auto &it : GameData::Start())
+	{
+		Rectangle zone(
+			point  + Point(
+				(entryInternalBox.Width()) / 2, 
+				(entryInternalBox.Height()) / 2), 
+			entryBox.Dimensions()
+		);
+		if(point.Y() > entryListBox.Bottom() || point.Y() < entryListBox.Top())
+		{
+			// Don't bother drawing if the item is above or under the list
+			point += Point(0., entryBox.Height());
+			continue;
+		}
+		
+		
+		bool isHighlighted = (it == chosenStart);
+		
+		// double alpha = min(1., max(0., min(.1 * (113. - point.Y()), .1 * (point.Y() - -167.))));
+		double alpha = 1;
+		
+		if(it == chosenStart)
+		{
+			FillShader::Fill(zone.Center(), zone.Dimensions(), Color(.1 * alpha, 0.));
+		}
+		
+		DisplayText name = DisplayText(it.GetName(), Truncate::BACK);
+		font.Draw(name, point, Color((isHighlighted ? .7 : .5) * alpha, 0.));
+		point += Point(0., entryBox.Height());
+	}
+}
+
+
+
+bool StartConditionsPanel::Drag(double dx, double dy)
+{
+	int x = hoverPoint.X();
+	if(x >= entryListBox.Left() && x < entryListBox.Right())
+	{
+		// Scroll the list
+		// This looks inefficient but it probably gets optimized by the compiler
+		listScroll -= dy;
+		listScroll = min(entryBox.Height() * (GameData::Start().size()-1), listScroll); // Avoid people going too low
+		listScroll = max(0., listScroll); // Snap the list to avoid people scrolling too far up
+	}
+	else
+	{
+		// Scroll the description
+		descriptionScroll += dy;
+		descriptionScroll = min(0., descriptionScroll); // Snap the list to avoid people scrolling too far up
```

Comment:
Inline / non-sentence comments don't meet the style guide. It's generally sufficient to annotate the block with the intent, rather than the play-by-play. If the code itself is not clear, then more verbose comments are warranted.